src/ShallowWrapper.lua

```diff
@@ -0,0 +1,228 @@
+local Children = require(script.Parent.PropMarkers.Children)
+local ElementKind = require(script.Parent.ElementKind)
+local ElementUtils = require(script.Parent.ElementUtils)
+local snapshot = require(script.Parent.snapshot)
+
+local ShallowWrapper = {}
+local ShallowWrapperMetatable = {
+	__index = ShallowWrapper,
+}
+
+local function getTypeFromVirtualNode(virtualNode)
+	local element = virtualNode.currentElement
+	local kind = ElementKind.of(element)
+
+	if kind == ElementKind.Host then
+		return {
+			kind = ElementKind.Host,
+			className = element.component,
+		}
+	elseif kind == ElementKind.Function then
+		return {
+			kind = ElementKind.Function,
+			functionComponent = element.component,
+		}
+	elseif kind == ElementKind.Stateful then
+		return {
+			kind = ElementKind.Stateful,
+			component = element.component,
+		}
+	else
+		error(('shallow wrapper does not support element of kind %q'):format(tostring(kind)))
```

Comment:
reminder for us to document the types we support when we get to writing documentation for snapshot testing